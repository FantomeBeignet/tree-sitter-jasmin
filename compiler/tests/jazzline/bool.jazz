abstract predicate int b2i (bool);

fn _bool(reg u64 a) -> reg u64
ensures #[prover=smt] {result.0 == a}
{
reg u64 r;
reg bool cf;
cf, r = a + 1;
#[kind=Assert, prover=smt] assert ( cf == true);
#[kind=Assert, prover=cas] assert ( b2i(cf) == 1);
return r;
}
