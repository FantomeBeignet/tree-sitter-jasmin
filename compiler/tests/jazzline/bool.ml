open Common

let () = parse_and_print "bool.jazz"  "_bool"
