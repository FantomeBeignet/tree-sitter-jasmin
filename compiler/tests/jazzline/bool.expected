proc _bool(r_152@uint64, a_137@uint64) =
{
 true && true
}
adds cf_151@uint1 r_152@uint64 a_137@uint64 (1)@uint64;
assert true && (cf_151@uint1 = (const 1 (1)));
assert /\[(cf_151 = (1))] && true;
{
 true && (r_152@uint64 = a_137@uint64)
}
 